### PR TITLE
SBP add category model

### PR DESCRIPTION
**Pull request by petro2061**
Implemented a new controller containing category management endpoints. The entity of the category represents a certain type to which many books can belong. Therefore, a category field marked with an annotation @ManytoMany is added to the book entity. Also to project added category repository and category servise that include method for delete, save and update mathod for management category entity. To book reposetory and servise added method for find book by category id. Also implemented Category mapper class and modifired book mapper class. To mathod that search books by params adde new provider for serch by category id and modifired another providers for avoid lazy initialsation exseption.

### CODE REVIEW

File: src/main/java/project/onlinebookstore/dto/book/CreateBookRequestDto.java [hunk 1]
```diff
@@ -23,5 +24,7 @@ public class CreateBookRequestDto {
     private String description;
     @CoverImage
     private String coverImage;
+    @NotNull(message = "Can be empty but not null")
```
Comment (Elena-Bruyako) [inline src/main/java/project/onlinebookstore/dto/book/CreateBookRequestDto.java]:
```suggestion
    @NotEmpty(message = "Can be empty but not null")
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Just one comment

**Comment by petro2061** [comment]
Changed annotation NotNull to NotEmpty

**Review by Elena-Bruyako — APPROVED** [review]
well done

**Pull request merged**